### PR TITLE
Create usage rights from copyright group

### PR DESCRIPTION
**Pull request by jamesgorrie**
@theefer 

Seems a good place to start - I am then assuming we can override this information in the edits service.

### CODE REVIEW

File: picdar-export/src/main/scala/com.gu.mediaservice.picdarexport/lib/cleanup/UsageRightsOverride.scala [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  def commissionedPhotographer(m: ImageMetadata) =
+    CommissionedPhotographer(removeCommissioned(m.copyright.get), getPublication(m.copyright.get))
+
+  // TODO: What do we do with Commissioned Agencies
+  // e.g. "Copyright Group" == "Agencies - commissioned"
+  val copyrightGroupToUsageRightsMap: Map[String, (ImageMetadata) => UsageRights] =
+    Map(
+      "Free images" -> handout,
+      "TV / film publicity images" -> handout,
+      "TV / film publicity images 2" -> handout,
```
Comment (theefer) [inline picdar-export/src/main/scala/com.gu.mediaservice.picdarexport/lib/cleanup/UsageRightsOverride.scala]: Not PR?

File: picdar-export/src/main/scala/com.gu.mediaservice.picdarexport/lib/cleanup/UsageRightsOverride.scala [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+      "Freelance photographers - commissioned" -> commissionedPhotographer,
+      "Freelance Weekend/Labs photographers - commissioned" -> commissionedPhotographer,
+
+      "Agencies - contract Rex Features" -> ((m: ImageMetadata) => Agency("Rex Features")),
+      "Agencies - contract Barcroft Media" -> ((m: ImageMetadata) => Agency("Barcroft Media")),
+      // there is no consistent collections here, but it doesn't mater as they're all free to use
+      // and we only exclude from the collection property
+      "Agencies - contract Getty Collections" -> ((m: ImageMetadata) => Agency("Getty Images")),
+      "Agencies - contract Reuters" -> ((m: ImageMetadata) => Agency("Reuters")),
+      "Agencies - contract" -> ((m: ImageMetadata) => Agency(m.copyright.get)),
```
Comment (theefer) [inline picdar-export/src/main/scala/com.gu.mediaservice.picdarexport/lib/cleanup/UsageRightsOverride.scala]: `.get` can throw if Option is empty :(

File: picdar-export/src/main/scala/com.gu.mediaservice.picdarexport/lib/cleanup/UsageRightsOverride.scala [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+      "Readers pictures" -> ((m: ImageMetadata) => GuardianWitness()),
+      "Readers' pictures" -> ((m: ImageMetadata) => GuardianWitness())
```
Comment (theefer) [inline picdar-export/src/main/scala/com.gu.mediaservice.picdarexport/lib/cleanup/UsageRightsOverride.scala]: Is that actually true (i.e. all readers picture are witness ones)?

File: picdar-export/src/main/scala/com.gu.mediaservice.picdarexport/lib/cleanup/UsageRightsOverride.scala [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+      "Freelance photographers - contract" -> contractPhotographer,
+
+      "Freelance photographers - commissioned" -> commissionedPhotographer,
+      "Freelance Weekend/Labs photographers - commissioned" -> commissionedPhotographer,
+
+      "Agencies - contract Rex Features" -> ((m: ImageMetadata) => Agency("Rex Features")),
+      "Agencies - contract Barcroft Media" -> ((m: ImageMetadata) => Agency("Barcroft Media")),
+      // there is no consistent collections here, but it doesn't mater as they're all free to use
+      // and we only exclude from the collection property
+      "Agencies - contract Getty Collections" -> ((m: ImageMetadata) => Agency("Getty Images")),
```
Comment (theefer) [inline picdar-export/src/main/scala/com.gu.mediaservice.picdarexport/lib/cleanup/UsageRightsOverride.scala]: Presumably we could use the `metadata.source` here?

File: picdar-export/src/main/scala/com.gu.mediaservice.picdarexport/lib/cleanup/UsageRightsOverride.scala [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+object UsageRightsOverride {
+
+  def removeCommissioned(s: String) = s.replace("(commissioned)", "").trim
+  def getPublication(s: String) = PhotographersList.getPublication(MetadataConfig.allPhotographers, s).getOrElse("The Guardian")
+
+  def handout(m: ImageMetadata) = Handout()
+  def staffPhotographer(m: ImageMetadata) = StaffPhotographer(m.byline.get, m.copyright.get.toLowerCase match {
+    // we know these are the only values
+    case "the observer" => "The Observer"
```
Comment (theefer) [inline picdar-export/src/main/scala/com.gu.mediaservice.picdarexport/lib/cleanup/UsageRightsOverride.scala]: do we need `"observer"` too?

File: picdar-export/src/main/scala/com.gu.mediaservice.picdarexport/lib/cleanup/UsageRightsOverride.scala [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+    case "guardian" => "The Guardian"
+    case "for the guardian" => "The Guardian"
+    case "the guardian" => "The Guardian"
+    case _ => "The Guardian"
+  })
+
+  def contractPhotographer(m: ImageMetadata) =
+    ContractPhotographer(removeCommissioned(m.copyright.get), getPublication(m.copyright.get))
+  def commissionedPhotographer(m: ImageMetadata) =
+    CommissionedPhotographer(removeCommissioned(m.copyright.get), getPublication(m.copyright.get))
```
Comment (theefer) [inline picdar-export/src/main/scala/com.gu.mediaservice.picdarexport/lib/cleanup/UsageRightsOverride.scala]: `get` :-/

File: picdar-export/src/main/scala/com.gu.mediaservice.picdarexport/lib/cleanup/UsageRightsOverride.scala [hunk 7]
```diff
@@ -0,0 +1,55 @@
+package com.gu.mediaservice.picdarexport.lib.cleanup
+
+import com.gu.mediaservice.lib.config.{MetadataConfig, PhotographersList}
+import com.gu.mediaservice.model._
+
+object UsageRightsOverride {
+
+  def removeCommissioned(s: String) = s.replace("(commissioned)", "").trim
+  def getPublication(s: String) = PhotographersList.getPublication(MetadataConfig.allPhotographers, s).getOrElse("The Guardian")
```
Comment (theefer) [inline picdar-export/src/main/scala/com.gu.mediaservice.picdarexport/lib/cleanup/UsageRightsOverride.scala]: Jo OK with defaulting to The Guardian? Isn't publication optional on contract/commissioned photographers?

File: picdar-export/src/main/scala/com.gu.mediaservice.picdarexport/lib/cleanup/UsageRightsOverride.scala [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+      "Agencies - contract Barcroft Media" -> ((m: ImageMetadata) => Agency("Barcroft Media")),
+      // there is no consistent collections here, but it doesn't mater as they're all free to use
+      // and we only exclude from the collection property
+      "Agencies - contract Getty Collections" -> ((m: ImageMetadata) => Agency("Getty Images")),
+      "Agencies - contract Reuters" -> ((m: ImageMetadata) => Agency("Reuters")),
+      "Agencies - contract" -> ((m: ImageMetadata) => Agency(m.copyright.get)),
+
+      "Readers pictures" -> ((m: ImageMetadata) => GuardianWitness()),
+      "Readers' pictures" -> ((m: ImageMetadata) => GuardianWitness())
+    )
```
Comment (theefer) [inline picdar-export/src/main/scala/com.gu.mediaservice.picdarexport/lib/cleanup/UsageRightsOverride.scala]: Out of curiosity, are there many more group we're not supporting yet?

**Comment by theefer** [comment]
Nice! Some comments above.

Would we want to set byline/credit for photographers, or is it already set reliably enough?

File: picdar-export/src/main/scala/com.gu.mediaservice.picdarexport/lib/cleanup/UsageRightsOverride.scala (hunk 1 above)
Comment (jamesgorrie) [inline picdar-export/src/main/scala/com.gu.mediaservice.picdarexport/lib/cleanup/UsageRightsOverride.scala]: Ah yes

File: picdar-export/src/main/scala/com.gu.mediaservice.picdarexport/lib/cleanup/UsageRightsOverride.scala (hunk 2 above)
Comment (jamesgorrie) [inline picdar-export/src/main/scala/com.gu.mediaservice.picdarexport/lib/cleanup/UsageRightsOverride.scala]: With all of these, the copyright will be set s that's how you set the copyright group.
Happy to make allowances for this if you think it's worth it (this is just slightly clearer).

File: picdar-export/src/main/scala/com.gu.mediaservice.picdarexport/lib/cleanup/UsageRightsOverride.scala (hunk 3 above)
Comment (jamesgorrie) [inline picdar-export/src/main/scala/com.gu.mediaservice.picdarexport/lib/cleanup/UsageRightsOverride.scala]: Yip - reference the doc I sent.

File: picdar-export/src/main/scala/com.gu.mediaservice.picdarexport/lib/cleanup/UsageRightsOverride.scala (hunk 5 above)
Comment (jamesgorrie) [inline picdar-export/src/main/scala/com.gu.mediaservice.picdarexport/lib/cleanup/UsageRightsOverride.scala]: These values are hard coded - reference the doc....

File: picdar-export/src/main/scala/com.gu.mediaservice.picdarexport/lib/cleanup/UsageRightsOverride.scala (hunk 8 above)
Comment (jamesgorrie) [inline picdar-export/src/main/scala/com.gu.mediaservice.picdarexport/lib/cleanup/UsageRightsOverride.scala]: Yes, there is one I am trying to think of what to do with which is commissioned agencies.
These are images we commission photographs off or potentially buy in. Good to have a chat about it.

I haven't had a massive look at how many images there are in this category.

Comment (theefer) [inline picdar-export/src/main/scala/com.gu.mediaservice.picdarexport/lib/cleanup/UsageRightsOverride.scala]: This sounds like the "Acquired"/"Paid for" category we'd mentioned a while back, i.e. we've shelled out and paid for that one so we can now use it. Worth talking to Robert/Jo about these though as I don't know if they'd allow repeated use, always or sometimes, etc.

File: picdar-export/src/main/scala/com.gu.mediaservice.picdarexport/lib/cleanup/UsageRightsOverride.scala (hunk 5 above)
Comment (theefer) [inline picdar-export/src/main/scala/com.gu.mediaservice.picdarexport/lib/cleanup/UsageRightsOverride.scala]: OK

**Comment by jamesgorrie** [comment]
> Would we want to set byline/credit for photographers, or is it already set reliably enough?

I think this is something that is going to have to be handled by the edits service.
i.e.
```
POST .../usage-rights/iouoiwquroiwer -d
{
  "category": "staff-photographer",
  "photographer": "Hamah Bradburn",
  "publication": "The Guardian"
}
```

We should then not only set the usage rights to the above, but override the metadata with:
```JSON
{
  "credit": "The Guardian",
  "byline": "Hamzah Bradburn"
}
"""

as I have seen this catch a few people out already.

**Comment by theefer** [comment]
You mean have the metadata service alter the metadata as a side-effect of updating the rights?

**Comment by jamesgorrie** [comment]
Something like that.

**Comment by theefer** [comment]
Not keen on side-effects as it makes it harder to code the UI and hard/impossible to build the UX we want on top if the user wants to opt out of the behaviour. Spurious example: if you've just edited the credit or byline, maybe you know better and the rights shouldn't reset what you've just done. Letting the UI decide when to apply these rules gives more control to the UI. Or in this case, simply the export script.

But I'd leave that idea aside for now as that branch is already useful with just the rights.

File: picdar-export/src/main/scala/com.gu.mediaservice.picdarexport/lib/cleanup/UsageRightsOverride.scala (hunk 2 above)
Comment (theefer) [inline picdar-export/src/main/scala/com.gu.mediaservice.picdarexport/lib/cleanup/UsageRightsOverride.scala]: Unless it makes our code a lot worse, I'm not too keen to forego type safety by relying on Picdar's (of all systems!) strictness in applying its own copyright model and not having crappy data here and there.

**Comment by jamesgorrie** [comment]
Yip, sorry, I was off topic.
Thought the UI would be a good place e.g. Apply usage rights and surface a "apply to metadata" button.

In the case of this branch, as the data has been edited / overridden already, I think we should leave this as overriding the usageRights.

File: picdar-export/src/main/scala/com.gu.mediaservice.picdarexport/lib/cleanup/UsageRightsOverride.scala (hunk 8 above)
Comment (jamesgorrie) [inline picdar-export/src/main/scala/com.gu.mediaservice.picdarexport/lib/cleanup/UsageRightsOverride.scala]: I've spoke to Jo about them quite a lot - let's talk :panda_face: 2 :dragon_face:

File: picdar-export/src/main/scala/com.gu.mediaservice.picdarexport/lib/cleanup/UsageRightsOverride.scala (hunk 7 above)
Comment (jamesgorrie) [inline picdar-export/src/main/scala/com.gu.mediaservice.picdarexport/lib/cleanup/UsageRightsOverride.scala]: No it's not - this is a larger conversation about what publication is.
For now this is fine.

File: picdar-export/src/main/scala/com.gu.mediaservice.picdarexport/lib/cleanup/UsageRightsOverride.scala (hunk 6 above)
Comment (jamesgorrie) [inline picdar-export/src/main/scala/com.gu.mediaservice.picdarexport/lib/cleanup/UsageRightsOverride.scala]: [Removed](https://github.com/guardian/grid/commit/7ccc73600dbb62a512fcee924590ca84e417e441).

File: picdar-export/src/main/scala/com.gu.mediaservice.picdarexport/lib/cleanup/UsageRightsOverride.scala (hunk 4 above)
Comment (jamesgorrie) [inline picdar-export/src/main/scala/com.gu.mediaservice.picdarexport/lib/cleanup/UsageRightsOverride.scala]: As we have a list of sources, and we know that these aren't excluded, I don't see why we would, given that a lot of past data is gumpf.

File: picdar-export/src/main/scala/com.gu.mediaservice.picdarexport/lib/cleanup/UsageRightsOverride.scala (hunk 2 above)
Comment (jamesgorrie) [inline picdar-export/src/main/scala/com.gu.mediaservice.picdarexport/lib/cleanup/UsageRightsOverride.scala]: [It's more verbose, but safer](https://github.com/guardian/grid/commit/7ccc73600dbb62a512fcee924590ca84e417e441)

File: picdar-export/src/main/scala/com.gu.mediaservice.picdarexport/lib/cleanup/UsageRightsOverride.scala [hunk 9]
```diff
@@ -0,0 +1,70 @@
+package com.gu.mediaservice.picdarexport.lib.cleanup
+
+import com.gu.mediaservice.lib.config.{MetadataConfig, PhotographersList}
+import com.gu.mediaservice.model._
+
+object UsageRightsOverride {
+
+  // scared of typos
+  val theGuardian = "The Guardian"
+  val theObserver = "The Guardian"
```
Comment (theefer) [inline picdar-export/src/main/scala/com.gu.mediaservice.picdarexport/lib/cleanup/UsageRightsOverride.scala]: ?

File: picdar-export/src/main/scala/com.gu.mediaservice.picdarexport/lib/cleanup/UsageRightsOverride.scala [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+  def removeCommissioned(s: String) = s.replace("(commissioned)", "").trim
+  def getPublication(s: String) = PhotographersList.getPublication(MetadataConfig.allPhotographers, s).getOrElse(theGuardian)
+
+  def prImage(m: ImageMetadata) = Some(PrImage())
+  def guardianWitness(m: ImageMetadata) = Some(GuardianWitness())
+  def agency(agency: String) = Some(Agency(agency))
+
+  def handout(m: ImageMetadata) = Some(Handout())
+  def staffPhotographer(m: ImageMetadata) = (m.byline, m.copyright.map(_.toLowerCase)) match {
+    case (Some(byline), Some("the observer"))     => Some(StaffPhotographer(byline, "The Observer"))
```
Comment (theefer) [inline picdar-export/src/main/scala/com.gu.mediaservice.picdarexport/lib/cleanup/UsageRightsOverride.scala]: `theObserver`?

Comment (jamesgorrie) [inline picdar-export/src/main/scala/com.gu.mediaservice.picdarexport/lib/cleanup/UsageRightsOverride.scala]: :eyes: thanks.

**Comment by jamesgorrie** [comment]
@theefer after chatting to Jo, I've found some more category groups, namely "free images" and "agency commissioned".

I've added [those](https://github.com/guardian/grid/commit/3ae7f07837e7a87b9345bf65029958e541535af3) [two](https://github.com/guardian/grid/commit/a809964214a3608ce364c2235f13c626d26866db) now.

**Comment by theefer** [comment]
:+1:

**Comment by jamesgorrie** [comment]
We'l work on the "doing" of this next.

**Pull request merged**